themes/10up-theme/includes/blocks.php

```diff
@@ -42,19 +42,49 @@ function setup() {
 }
 
 /**
- * Add in blocks that are registered in this theme
+ * Automatically registers all blocks that are located within the includes/blocks directory
  *
  * @return void
  */
 function register_theme_blocks() {
 	// Filter the plugins URL to allow us to have blocks in themes with linked assets. i.e editorScripts
 	add_filter( 'plugins_url', __NAMESPACE__ . '\filter_plugins_url', 10, 2 );
 
-	// Require custom blocks.
-	require_once TENUP_THEME_BLOCK_DIR . '/example-block/register.php';
+	// Register blocks for the active theme, for both parent and child theme,
+	// if applicable.
+	foreach ( wp_get_active_and_valid_themes() as $theme ) {
```

Comment:
Noting that `wp_get_active_and_valid_themes()` is for core use only.

https://developer.wordpress.org/reference/functions/wp_get_active_and_valid_themes/

Reply:
Ressolved (y)